zenith/src/main.rs

```diff
@@ -178,12 +182,86 @@ fn main() -> Result<()> {
     Ok(())
 }
 
+// Print branches list as a tree-like structure.
+fn print_branches_tree(branches: Vec<BranchInfo>) {
+    let mut branches_hash: HashMap<String, BranchTreeEl> = HashMap::new();
+
+    // Form a hash table of branch timeline_id -> BranchTreeEl.
+    for branch in &branches {
+        branches_hash.insert(
+            branch.timeline_id.to_string(),
+            BranchTreeEl {
+                info: branch.clone(),
+                children: Vec::new(),
+            },
+        );
+    }
+
+    // Memorize all direct children of each branch.
+    for branch in &branches {
+        if let Some(name) = &branch.ancestor_id {
+            branches_hash
+                .get_mut(name)
+                .unwrap()
```

Comment:
unwrap -> ?

Reply:
OK, it wasn't that straightforward. We couldn't use just ? here, since this function wasn't returning a Result. And even with returning a result it is not possible to use ? without wrapping with additional context. I'm not a true rustacean to tell why exactly things work this way, so I've just refactored this piece of code a bit